vendor/mcmc/mcmc.c

```diff
@@ -358,32 +459,272 @@ static int _mcmc_parse_response(mcmc_ctx_t *ctx, mcmc_resp_t *r) {
     }
 }
 
-// EXTERNAL API
+// TODO: possible to codegen the 32 vs 64 variants
```

Comment:
need to at least add a note here: not defending against len == 0 in these functions.

the while loop won't do anything but the minus check can go past if the len is 0. There's no practical case where a token can exist and +1'ing would go past the end of the buffer though: we _must_ have a `\n` at the end of the string.

So I should probably add this as a comment and see about adjusting later.